src/command/stash.ts

```diff
@@ -0,0 +1,78 @@
+import { git } from "../core";
+
+export namespace stash {
+    /**
+     * Stash the current changes.
+     *
+     * @param repositoryPath the repository path to the local Git clone.
+     * @param message an optional stash message.
+     */
+    export async function push(repositoryPath: string, message?: string): Promise<void> {
+        const args: string[] = ['stash', 'push'];
+        if (message) {
+            args.push('-m', message);
+        }
+        await git(args, repositoryPath, 'stash');
+    }
+
+    /**
+     * List the current stashes.
+     *
+     * @param repositoryPath the repository path to the local Git clone.
+     */
+    export async function list(repositoryPath: string): Promise<string[]> {
```

Comment:
Please adjust each function that it has the optional `options?: IGitExecutionOptions` argument.

Otherwise, we won't be able to use the same Git commands over SSH. Check other Git commands; `log`, `pull`, or `commit`.